corehq/util/pagination.py

```diff
@@ -55,14 +62,149 @@ def paginate_function(get_data_fn, next_args_fn, event_handler=PaginationEventHa
     while len_results:
         event_handler.page_start(total_emitted, *args, **kwargs)
         results = get_data_fn(*args, **kwargs)
-        start_time = datetime.datetime.utcnow()
+        start_time = datetime.utcnow()
         len_results = len(results)
 
         for item in results:
             yield item
 
         total_emitted += len_results
-        event_handler.page_end(total_emitted, datetime.datetime.utcnow() - start_time, *args, **kwargs)
+        event_handler.page_end(total_emitted, datetime.utcnow() - start_time, *args, **kwargs)
 
         if len_results:
             args, kwargs = next_args_fn(item, *args, **kwargs)
+
+
+class ResumableIteratorState(JsonObject):
+    doc_type = "ResumableIteratorState"
+    _id = StringProperty()
+    name = StringProperty()
+    timestamp = DateTimeProperty()
+    args = ListProperty()
+    kwargs = DictProperty()
+    retry = DictProperty()
+    progress = DictProperty()
```

Comment:
Yes it does but I can change it to a `DefaultProperty` which is unconstrained. I actually prefer having it be a dict though since then you always know what you're going to get back. It also gives you more information about what the data in the progress is since each element would need a key which also serves as a descriptor., You can still store anything as a value so it doesn't restrict the options.